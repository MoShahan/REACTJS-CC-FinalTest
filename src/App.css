form {
    display: flex;
    justify-content: center;
    align-items: center;
    margin: 50px 0;
    gap: 2rem;
}