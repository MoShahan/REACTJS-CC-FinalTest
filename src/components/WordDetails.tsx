import { Paper } from "@mui/material";
import React from "react";
import ReactAudioPlayer from "react-audio-player";

const WordDetails = ({ displayWord }: any) => {
  // console.log(displayWord);

  return (
    <div>
      <Paper
        elevation={10}
        sx={{
          display: "flex",
          alignItems: "center",
          justifyContent: "center",
          flexDirection: "column",
          width: "50%",
          margin: "0 auto",
          bgcolor: "lightblue",
        }}
      >
        <h3> {displayWord.word} </h3>
        <h3> {displayWord.phonetic} </h3>
      </Paper>
      <Paper
        elevation={10}
        sx={{
          display: "flex",
          alignItems: "center",
          justifyContent: "center",
          flexDirection: "column",
          // width: "500px",
          margin: "auto",
          padding: "5px 5px",
          marginTop: "20px",
        }}
      >
        {displayWord?.meanings.map((meaning: any, index: number) => (
          <div key={index}>
            <h3>{meaning.partOfSpeech}</h3>
            {meaning.definitions.map((definition: any, i: number) => (
              <div key={index + i}>
                <hr />
                <h5>{definition.definition}</h5>
                <h5>Eg: {definition.example}</h5>
              </div>
            ))}
          </div>
        ))}
      </Paper>
      <Paper
        elevation={10}
        sx={{
          display: "flex",
          alignItems: "center",
          justifyContent: "center",
          flexDirection: "column",
          // width: "500px",
          margin: "auto",
          padding: "10px 50px",
          marginTop: "20px",
        }}
      >
        <ReactAudioPlayer
          src={displayWord?.phonetics[0].audio}
          autoPlay
          controls
        />
      </Paper>
    </div>
  );
};

export default WordDetails;
