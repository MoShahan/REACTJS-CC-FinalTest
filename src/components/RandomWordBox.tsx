import { Button, Paper } from "@mui/material";
import React, { useEffect, useState } from "react";
import WordDetails from "./WordDetails";
import axios from "axios";
import { DICTIONARY_API } from "../App";

const RandomWordBox = ({ word, definition }: any) => {
  //   console.log("Random Word ===", word);
  const [showRDetails, setShowRDetails] = useState<boolean>(false);
  const [thisWordDetails, setThisWordDetails] = useState<Array<any>>([]);

  useEffect(() => {
    axios
      .get(DICTIONARY_API + word)
      .then((res: any) => {
        // setLoading(false);
        setThisWordDetails(res.data);
      })
      .catch((e) => console.error(e));
  }, []);

  return (
    <Paper
      elevation={10}
      sx={{
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        flexDirection: "column",
        width: "500px",
        margin: "auto",
        padding: "10px 50px",
      }}
    >
      {!showRDetails ? (
        <>
          <h1>Word of the Day</h1>
          <h2>Word : {word}</h2>
          <h2>Definition : {definition}</h2>
          <Button onClick={() => setShowRDetails(true)} variant="contained">
            SHOW MORE DETAILS
          </Button>
        </>
      ) : (
        <WordDetails displayWord={thisWordDetails} />
      )}
    </Paper>
  );
};

export default RandomWordBox;
