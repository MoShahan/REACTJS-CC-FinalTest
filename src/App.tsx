import React from "react";
import "./App.css";
import User from "./pages/User";

export const RANDOM_WORD_API = "https://random-words-api.vercel.app/word";
export const DICTIONARY_API =
  "https://api.dictionaryapi.dev/api/v2/entries/en/";

function App() {
  console.log("App has Rendered");

  return (
    <>
      <User />
    </>
  );
}

export default App;
