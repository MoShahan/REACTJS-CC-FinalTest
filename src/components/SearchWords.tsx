import { Button, Paper } from "@mui/material";
import React, { useState } from "react";
import { WordType } from "../pages/User";
import WordDetails from "./WordDetails";

type SearchWordsProps = {
  words: Array<WordType>;
};

const SearchWords = ({ words }: SearchWordsProps) => {
  const [showDetails, setShowDetails] = useState<boolean>(false);
  const [currWordDetails, setCurrWordDetails] = useState<any>();
  //   console.log("SearchWords has been rendered");
  //   console.log(words);

  return (
    <Paper
      sx={{
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        flexDirection: "column",
        width: "500px",
        margin: "auto",
        padding: "10px 50px",
      }}
      elevation={10}
    >
      {!showDetails ? (
        <div>
          {words?.map((word: any, index: any) => (
            <div key={index}>
              <h1>{word.word}</h1>
              <h3>{word?.meanings[0]?.definitions[0].definition}</h3>
              <Button
                variant="contained"
                onClick={(e) => {
                  setShowDetails(true);
                  setCurrWordDetails(word);
                }}
              >
                More Details
              </Button>
              <hr />
            </div>
          ))}
        </div>
      ) : (
        <div>
          <WordDetails displayWord={currWordDetails} />
        </div>
      )}
    </Paper>
  );
};

export default SearchWords;
