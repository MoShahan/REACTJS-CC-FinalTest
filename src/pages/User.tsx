import { Button, CircularProgress, TextField } from "@mui/material";
import axios from "axios";
import React, { useCallback, useEffect, useState } from "react";
import { DICTIONARY_API, RANDOM_WORD_API } from "../App";
import RandomWordBox from "../components/RandomWordBox";
import SearchWords from "../components/SearchWords";

const User = () => {
  const [word, setWord] = useState<string>("");
  const [wordDef, setWordDef] = useState<string>("");
  const [details, setDetails] = useState<any>();
  const [loading, setLoading] = useState<boolean>(true);
  const [searchWord, setSearchWord] = useState<string>("");
  const [isRandomWord, setIsRandomWord] = useState<boolean>(true);
  const [searchResults, setSearchResults] = useState<Array<any>>([]);

  useEffect(() => {
    axios
      .get(RANDOM_WORD_API)
      .then((res: any) => {
        console.log(res);
        setWord(res.data[0].word);
        setWordDef(res.data[0].definition);
        setLoading(false);
      })
      .catch((e) => {
        setLoading(false);
        console.error(e);
      });
  }, []);

  const handleSubmit = useCallback(
    (e: any) => {
      e.preventDefault();
      setIsRandomWord(false);
      setLoading(true);
      axios
        .get(DICTIONARY_API + searchWord)
        .then((res: any) => {
          setLoading(false);
          setSearchResults(res.data);
        })
        .catch((e) => console.error(e));
    },
    [searchWord]
  );

  return (
    <div>
      <form action="" onSubmit={handleSubmit}>
        <TextField
          autoFocus
          placeholder="Enter Keyword"
          value={searchWord}
          onChange={(e) => setSearchWord(e.target.value)}
        />
        <Button
          sx={{ transform: "scale(1.5)" }}
          type="submit"
          variant="contained"
        >
          SEARCH
        </Button>
      </form>
      <hr />
      {loading ? (
        <CircularProgress />
      ) : isRandomWord ? (
        <div>
          <RandomWordBox
            word={word}
            // onClick={getMoreDetails}
            definition={wordDef}
          />
        </div>
      ) : (
        <div>
          <SearchWords words={searchResults} />
        </div>
      )}
      <hr />
    </div>
  );
};

export default User;
