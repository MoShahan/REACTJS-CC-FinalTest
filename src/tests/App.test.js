import React from "react";
import { act, fireEvent, render, screen } from "@testing-library/react";
import User from "../pages/User";
import WordDetails from "../components/WordDetails";
import SearchWords from "../components/SearchWords";
import ReactDOM from "react-dom/client";
import RandomWordBox from "../components/RandomWordBox";
import axios from "axios";
import { displayWordProps, randomProps, searchWordsProps } from "./testingVariables";
import App from "../App";

describe("Rendering All Components...", () => {
  test("App...", () => {
    render(<App />);
  });

  test("User...", () => {
    render(<User />);
  });

  test("WordDetails...", () => {
    render(<WordDetails displayWord={displayWordProps} />);
  });

  test("SearchWords...", () => {
    render(<SearchWords words={searchWordsProps} />);
  });

  test("RandomWordBox...", () => {
    render(
      <RandomWordBox
        word={randomProps.word}
        definition={randomProps.definition}
      />
    );
  });
});

describe("Button Clicks...", () => {
  test("Search Button...", () => {
    render(<User />);
    const searchBtn = screen.getByText("SEARCH");
    fireEvent.click(searchBtn);
  });

  // test("Details Button...", () => {
  //   render(<WordDetails displayWord={displayWordProps} />);
  //   const detailsBtn = screen.getByText("More Details");
  //   fireEvent.click(detailsBtn);
  // });
});

describe("Input Elements...", () => {
  test("Search Input", () => {
    render(<User />);
    const inpt = screen.getByPlaceholderText("Enter Keyword");
    fireEvent.change(inpt, { target: { value: "word" } });
  });
});

describe("Snapshot Testing...", () => {
  test("App Component...", () => {
    const { AppComp } = render(<App />);
    expect(AppComp).toMatchSnapshot();
  });
});

// test("Rendering Index.tsx file...", () => {
//   const rootDiv = document.createElement("div");
//   rootDiv.id = "root";
//   document.body.appendChild(rootDiv);
//   require("./index.tsx");
// });
