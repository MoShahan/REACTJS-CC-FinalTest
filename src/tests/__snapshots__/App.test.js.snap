// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing... App Component... 1`] = `undefined`;
